src/sentry/static/sentry/app/views/inviteMember/inviteMember.jsx

```diff
@@ -0,0 +1,228 @@
+
+  getInitialState() {
+    let {teams} = this.getOrganization();
+
+    //select team if there's only one
+    let initialTeamSelection = teams.length === 1 ? [teams[0].id] : [];
+
+    return {
+      selectedTeams: new Set(initialTeamSelection),
+      isInvite: undefined,
+      roleList: [],
+      selectedRole: 'member',
+      email: '',
+      loading: true,
+      error: undefined
+    };
+  },
+
+  componentDidMount() {
+    let {slug} = this.getOrganization();
+    this.api.request(`/organizations/${slug}/members/me/`, {
+      method: 'GET',
+      success: ({role_list, is_invite}) => {
+        this.setState({roleList: role_list, isInvite: is_invite, loading: false});
+        if (role_list.filter(({_, allowed}) => allowed).length === 0) {
+          //not allowed to invite, redirect
+          this.redirectToMemberPage();
+        }
+      },
+      error: error => {
+        Raven.captureMessage('data fetch error ', {
+          extra: {error, state: this.state}
+        });
+      }
+    });
+  },
+
+  redirectToMemberPage() {
+    let {slug} = this.getOrganization();
+    window.location.href = `/organizations/${slug}/members/`;
+  },
+
+  splitEmails(text) {
+    return text.split(',').map(e => e.trim()).filter(e => e);
+  },
+
+  inviteUser(email) {
+    let {slug} = this.getOrganization();
+    let {selectedTeams, selectedRole} = this.state;
+
+    return new Promise((resolve, reject) => {
+      this.api.request(`/organizations/${slug}/members/`, {
+        method: 'POST',
+        data: {
+          email,
+          user: email,
+          teams: Array.from(selectedTeams.keys()),
+          role: selectedRole
+        },
+        success: () => {
+          AlertActions.addAlert({
+            message: `Added ${email}`,
+            type: 'success'
+          });
+          resolve();
+        },
+        error: err => {
+          if (err.responseJSON && err.responseJSON.exists) {
+            AlertActions.addAlert({
+              message: `User already exists: ${email}`,
+              type: 'info'
+            });
+            resolve();
+          } else {
+            reject(err.responseJSON);
+          }
+        }
+      });
+    });
+  },
+
+  submit() {
+    let {email} = this.state;
+    let emails = this.splitEmails(email);
+    if (!emails.length) return;
+    this.setState({loading: true});
+
+    Promise.all(emails.map(this.inviteUser))
+      .then(() => this.redirectToMemberPage())
+      .catch(error => {
+        if (!error.email && !error.role) {
+          Raven.captureMessage('unkown error ', {
+            extra: {error, state: this.state}
+          });
+        }
+        this.setState({error, loading: false});
+      });
+  },
+
+  toggleID(id) {
+    let {selectedTeams} = this.state;
+    if (selectedTeams.has(id)) {
+      selectedTeams.delete(id);
+    } else {
+      selectedTeams.add(id);
+    }
+    this.setState({selectedTeams});
+  },
+
+  renderRoleSelect() {
```

Comment:
It would be nice if we refactored this into a new component that accepted an initial data, reason being this will be used for member details.